src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,16 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.Warn("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Message types used for publishing should not implement interfaces.");
+
+                if (interfaces.Count > 1)
+                {
+                    Logger.ErrorFormat("Publishing type has multiple interfaces. Interfaces are {0}, the one that will be used for building routing key is {1}", string.Join(", ", interfaces), implementedInterface);
```

Comment:
@adamralph I suppose I can get on board with that reasoning...

It still feels like we'd want to put as much attention to this scenario as possible though, since you could consider it a message-loss scenario.